fabric-transaction-api-v1/src/main/java/net/fabricmc/fabric/api/transaction/v1/TransactionTracker.java

```diff
@@ -0,0 +1,24 @@
+package net.fabricmc.fabric.api.transaction.v1;
+
+public class TransactionTracker {
+	public static final TransactionTracker INSTANCE = new TransactionTracker();
```

Comment:
I hear weird multithreading noises

Reply:
I didn't see it necessary to add any sort of multithreading checks because the rest of the world isn't multithreaded either.

Reply:
No, there is client and server threads. You shouldn't make this a singleton! If fluid use it fluid has their own transaction handler, and same goes for other single usages.

Reply:
I personally believe we should bind a tracker to each thread.

So maybe the client get's it's own tracker, the server get's it's own and then we can have a spinoff `getInstance()` call that will obtain the tracker from it's current thread, and if one isn't on that thread, we create one for said thread.

Reply:
Oh right, didn't think of that. Whoops

Reply:
So how are you suggesting implementing this? Just a `ThreadLocal` or specifically have two fields for the client and server thread, and then a `ThreadLocal` for everything else? If so, how do I know I'm on the client or server thread?

Reply:
Shouldn't per thread, but just per usage, like one for event, one for fluid, etc.
thread specific isn't the safest either, and maybe you actually want atomic change to these transaction trackers.

Reply:
One for fluid would prevent using this to transfer anything else other than fluids in the same transaction.

What about having two default ones, a client and a server one, which you get by passing a world instance, and if any mod needs its own for other purposes then it can create another one itself.

Reply:
I don't think having a transaction per usage is useful. If someone starts an event transaction in the middle of a fluid transaction, the 2 transactions are necessary linked: aborting the fluid transaction should cancel the event transaction.

Reply:
I think what I have now should be sufficient for most use cases.